### DISCUSSION TITLE
Rich Text Editor Component

### DISCUSSION THREAD

**Discussion by sereneinserenade** [heroui-inc/heroui#206] [Ideas]
Next UI doesn't have a rich text-editor component right now. 

We could use https://tiptap.dev/ to create a rich text-editor component. Mantine uses QuillJS https://mantine.dev/others/rte/. I like Tiptap over QuillJS, for Tiptap can render React components https://tiptap.dev/guide/node-views/react.

I have no design for this, however provided that we have a design for this, I'd love to work on this.

**Comment by igniscyan** [comment]
Has there been any thought given to this yet? This seems to be a hole in NextUIs usability atm, Textareas don't seem to serve the same purpose for large amounts of text.

**Reply by LeeWio** [reply]
+1

**Comment by isaacdarcilla** [comment]
+1 on this.

**Reply by cozyCodr** [reply]
+1 on this in 2024

**Reply by mchrzanka** [reply]
shoot I was needing exactly this. +1

**Comment by sanjay10985** [comment]
I would like to do this with you, ready when you are

**Comment by IDJGILL** [comment]
+1

**Comment by creativekeyur** [comment]
+1

**Comment by vitaliyhan** [comment]
+1

**Comment by phemartin** [comment]
+1

**Comment by ojpetar** [comment]
+1

**Comment by TheDanniCraft** [comment]
+1

**Comment by dhananjaykumar880** [comment]
+1

**Comment by haidar47x** [comment]
+1

**Reply by PinkMeNow** [reply]
+1

**Comment by gltched-usr** [comment]
+1 But why do you guys still make yourself hope?

They rarely add new components but still try to convince you that HeroUi Pro is worth the heavy price.

(Spoiler: It's not. You get copy paste components that are hard to customize (Instead of bundling them into easy/ready to use components you need to copy and paste 20 files and to make changes you need to adjust the base files).

I really hoped this library would be better than mantine but sadly no :(

**Comment by PinkMeNow** [comment]
+1
